### PR TITLE
Improve dark theme on Windows

### PR DESCRIPTION
**Pull request by Torbjorn-Svensson**
Switch to CTabFolder/CTabItem for newui.
This changeset preserved binary API (as per tooling) but essentially
changes the meaning of some fields.

On a review of existing adopters/extenders of CDT no one is actually
accessing these fields.

Change-Id: Ic1ef0f242c1d2932726f3a6d4c9df9558312764a
Signed-off-by: Torbjörn Svensson <torbjorn.svensson@st.com>
Also-by: Jonah Graham <jonah@kichwacoders.com>

### CODE REVIEW

**Comment by Torbjorn-Svensson** [comment]
This is an import of https://git.eclipse.org/r/c/cdt/org.eclipse.cdt/+/192656

**Comment by Torbjorn-Svensson** [comment]
@Jonah, how do you want to proceed on this one?
Should we keep the old interface or should we just do the breaking change directly without caring about implementations that might extend the public interlace (that was never intended to be public...)?

File: core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java [hunk 1]
```diff
@@ -204,11 +206,13 @@ public abstract class AbstractPage extends PropertyPage implements IPreferencePa
 	 */
 	@Deprecated(forRemoval = true)
 	protected TabFolder folder;
```
Comment (jonahgraham) [inline core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java:208]: I think we can remove the `@Deprecated` fields now.

**Comment by jonahgraham** [comment]
> @jonah, how do you want to proceed on this one? Should we keep the old interface or should we just do the breaking change directly without caring about implementations that might extend the public interlace (that was never intended to be public...)?

I think as we should just do the breaking change.

PS I am @jonahgraham on GitHub (not `@jonah`)

**Review by jonahgraham — CHANGES_REQUESTED** [review]

**Review by jonahgraham — COMMENTED** [review]
This looks good to me - please also update https://github.com/eclipse-cdt/cdt/blob/main/doc/org.eclipse.cdt.doc.isv/guide/deprecated_API_removals.html

**Comment by Torbjorn-Svensson** [comment]
> I think as we should just do the breaking change.

Done in the last push.
I reverted to the old names to reduce the number of lines changed.

> PS I am @jonahgraham on GitHub (not `@jonah`)

Oops!

**Comment by jonahgraham** [comment]
@Torbjorn-Svensson I added a couple of extra commits, one to remove the rest of the fields from API that we had previously announced and one to update the removals doc. Please review and let me know what you think.

**Comment by jonahgraham** [comment]
:-( Sadly because you created this PR initially, I cannot add you as a reviewer. (BTW, note that I pushed my changes to the branch in your fork)

**Review by jonahgraham — APPROVED** [review]
@Torbjorn-Svensson assuming this passes the CI can you approve this.

File: core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java [hunk 2]
```diff
@@ -168,44 +168,22 @@ public abstract class AbstractPage extends PropertyPage implements IPreferencePa
...
+	 * @noreference This field is not intended to be referenced by clients.
 	 */
-	@Deprecated(forRemoval = true)
 	protected boolean noContentOnPage = false;
 	/**
-	 * @deprecated This field was never meant to be API. This field will be made private.
+	 * @noreference This field is not intended to be referenced by clients.
 	 */
-	@Deprecated(forRemoval = true)
 	protected boolean displayedConfig = false;
```
Comment (Torbjorn-Svensson) [inline core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java:177]: Shouldn't this field be private?

File: core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java [hunk 3]
```diff
@@ -168,44 +168,22 @@ public abstract class AbstractPage extends PropertyPage implements IPreferencePa
 	 * Bookeeping variables
 	 */
 	/**
-	 * @deprecated This field was never meant to be API. This field will be made private.
+	 * @noreference This field is not intended to be referenced by clients.
 	 */
-	@Deprecated(forRemoval = true)
 	protected boolean noContentOnPage = false;
```
Comment (Torbjorn-Svensson) [inline core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java:173]: Shouldn't this field be private?

File: core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java (hunk 2 above)
Comment (jonahgraham) [reply core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java:177]: Its accessed from AbstractSinglePage so I marked it as `@noreference` instead which gives us freedom to refactor in the future.

File: core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java (hunk 3 above)
Comment (jonahgraham) [reply core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractPage.java:173]: Its accessed from AbstractSinglePage so I marked it as `@noreference` instead which gives us freedom to refactor in the future.

**Comment by Torbjorn-Svensson** [comment]
> @Torbjorn-Svensson assuming this passes the CI can you approve this.

The PR creator can't approve this the normal way, but LGTM.

**Pull request merged**